src/osal/lv_freertos.c

```diff
@@ -261,8 +272,19 @@ lv_result_t lv_thread_sync_wait(lv_thread_sync_t * pxCond)
 lv_result_t lv_thread_sync_signal(lv_thread_sync_t * pxCond)
 {
 #if USE_FREERTOS_TASK_NOTIFY
-    /* Send a notification to the task waiting. */
-    xTaskNotifyGive(pxCond->xTaskToNotify);
+    _enter_critical();
+    TaskHandle_t task_to_notify = pxCond->xTaskToNotify;
+    pxCond->xTaskToNotify = NULL;
+    if(task_to_notify == NULL) {
+        /* No task waiting to be notified. Send this signal for later */
+        pxCond->xSignalSent = pdTRUE;
+    }
+    _exit_critical();
+
+    if(task_to_notify != NULL) {
```

Comment:
Please use the freertos naming convention: rename `task_to_notify` to 'xTaskToNotify`:

https://www.freertos.org/Documentation/02-Kernel/06-Coding-guidelines/02-FreeRTOS-Coding-Standard-and-Style-Guide#naming-conventions